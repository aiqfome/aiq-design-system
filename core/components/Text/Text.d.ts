import { ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps, TypographyProps } from 'styled-system';
export interface Props extends ColorProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps, TypographyProps {
    cursor?: string;
    truncate?: boolean;
    whiteSpace?: 'nowrap' | 'normal' | 'pre';
}
export declare const Text: import("styled-components").StyledComponent<"span", any, Props, never>;
//# sourceMappingURL=Text.d.ts.map