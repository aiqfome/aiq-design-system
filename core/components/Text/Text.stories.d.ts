import React from 'react';
declare const _default: {
    component: import("styled-components").StyledComponent<"span", any, import("./Text").Props, never>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const Basic: React.FC;
export declare const Truncated: React.FC;
//# sourceMappingURL=Text.stories.d.ts.map