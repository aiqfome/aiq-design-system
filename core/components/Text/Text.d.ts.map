{"version":3,"file":"Text.d.ts","sourceRoot":"","sources":["../../../src/components/Text/Text.tsx"],"names":[],"mappings":"AAEA,OAAO,EAEL,UAAU,EAEV,UAAU,EAEV,WAAW,EAEX,aAAa,EAEb,eAAe,EAEf,eAAe,EAChB,MAAM,eAAe,CAAA;AAEtB,MAAM,WAAW,KACf,SAAQ,UAAU,EAChB,UAAU,EACV,WAAW,EACX,aAAa,EACb,eAAe,EACf,eAAe;IACjB,MAAM,CAAC,EAAE,MAAM,CAAA;IACf,QAAQ,CAAC,EAAE,OAAO,CAAA;IAClB,UAAU,CAAC,EAAE,QAAQ,GAAG,QAAQ,GAAG,KAAK,CAAA;CACzC;AAED,eAAO,MAAM,IAAI,wEAuBhB,CAAA"}