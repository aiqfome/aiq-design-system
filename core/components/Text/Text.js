"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Text = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importStar(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.Text = styled_components_1.default.span `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.typography}

  white-space: ${props => props.whiteSpace || 'normal'};

  ${({ cursor }) => cursor &&
    styled_components_1.css `
      cursor: ${cursor};
    `};

  ${({ truncate }) => truncate &&
    styled_components_1.css `
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    `};
`;
exports.Text.propTypes = {
    cursor: prop_types_1.default.string,
    whiteSpace: prop_types_1.default.oneOf(['nowrap', 'normal', 'pre'])
};
