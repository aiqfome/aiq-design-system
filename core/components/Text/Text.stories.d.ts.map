{"version":3,"file":"Text.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Text/Text.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;;AAMzB,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAczB,CAAA;AAED,eAAO,MAAM,SAAS,EAAE,KAAK,CAAC,EAY7B,CAAA"}