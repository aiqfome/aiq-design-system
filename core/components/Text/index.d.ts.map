{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Text/index.tsx"],"names":[],"mappings":"AAAA,cAAc,QAAQ,CAAA"}