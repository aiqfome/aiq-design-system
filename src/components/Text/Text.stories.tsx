import React from 'react'
import { withKnobs, text, number } from '@storybook/addon-knobs'

import { Text } from './Text'
import { Flex } from '../Flex'

export default {
  component: Text,
  title: 'Text',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Text
    color={text('color', 'primary')}
    fontSize={number('fontSize', 16)}
    cursor={text('cursor', '')}
    fontWeight={text('fontWeight', 'medium')}
  >
    {text('text', 'Aiqfome')}
  </Text>
)

export const cursor: React.FC = () => (
  <Text cursor='pointer'>Design System</Text>
)

export const WhiteSpace: React.FC = () => (
  <Flex width='50px' border='1px solid black'>
    <Text whiteSpace='nowrap'>Design System</Text>
  </Flex>
)
