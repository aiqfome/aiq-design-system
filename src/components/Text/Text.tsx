import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import {
  color,
  ColorProps,
  space,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  typography,
  TypographyProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    TypographyProps {
  cursor?: string
  whiteSpace?: 'nowrap' | 'normal' | 'pre'
}

export const Text = styled.span<Props>`
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${typography}

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: ${cursor};
    `};
  white-space: ${props => props.whiteSpace || 'normal'};
`

Text.propTypes = {
  cursor: PropTypes.string,
  whiteSpace: PropTypes.oneOf(['nowrap', 'normal', 'pre'])
}
