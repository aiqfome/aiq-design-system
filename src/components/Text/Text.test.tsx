import React from 'react'

import { Text } from '../Text'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Text>Text component!</Text>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
