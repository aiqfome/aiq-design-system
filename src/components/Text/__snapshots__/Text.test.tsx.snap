// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<span
  className="sc-AxjAm jdFEEp"
>
  Text component!
</span>
`;
